### ISSUE TITLE
1.7.0 installer crashes immediately

### ISSUE THREAD

**Issue by yser23482hat** [Franiac/TwitchLeecher#373]
TLDR: Windows 7 SP1 64-bit, installer crashes and generates only one log file.
[log.txt](https://github.com/Franiac/TwitchLeecher/files/3447951/log.txt)

More details:

I had previous version of TL installed. Attempted to upgrade to 1.7.0 but installer hung for a very long time as it was installing .NET Framework 4.8, so I aborted it. Instead, I successfully installed .NET Framework 4.8 directly from Microsoft. Then I tried to install TL again but the installer crashed immediately.

The TL uninstaller in Windows Control Panel also crashed, so I used Revo Uninstaller to completely uninstall TL (it found and deleted files and registry keys). But even after that, TL 1.7.0 installer still crashes immediately (and so do installers of previous versions of TL).

Thank-you for reading!

**Comment by dmcrider** [comment]
I am able to replicate this issue - the installer hangs on installing .NET Framework 4.8 indefinitely

**Comment by Franiac** [comment]
Unfortunately the log des not help. you could try:

`TwitchLeecher_1.7.exe -l log.txt`

and copy ALL logs. There is one for the UI (log.txt), one for the .NET installer (*.html) and one for the MSI (log_0.txt or something). I need all of them. You only attached the UI log.

This is probably only a Win 7 problem since Win 10 >1607 already includes .NET 4.8 and won't run the installer.

**Comment by dmcrider** [comment]
I'm on Win10 Build 1809 and the .Net installer ran the first time I tried to install. After a computer restart it worked. See my edit above.

**Comment by Franiac** [comment]
@yser23482hat 

I have an additional question:
At what point during the installer process did the .NET 4.8 installer pop up?

1. Right when you started the TwitchLeecher_1.7.exe or
2. When you clicked through the TL installer dialog and finally clicked "Install".

This is important to know...

**Comment by Franiac** [comment]
@yser23482hat And what is a "very long time" in your eyes? The .NET installer can easily run 10min and more depending on the system.

**Comment by Franiac** [comment]
I have to correct my statement: .NET 4.8 is only preinstalled on Windows 10 1903

**Comment by yser23482hat** [comment]
Thank you for the responses both of you!

1) multiple restarts didn't solve the problem.

2) Only that one log was generated, there were no others (at least in the directory I ran the installer from).

3) The first time I tried it, the .NET installer popped up immediately, before anything else happened. Then after I aborted and installed .NET from Microsoft directly, the next time I tried the TL installer it no longer asked me to install .NET (but crashed as soon as the TL window came up). The logfile I posted is from this later installation attempt, not the original one I aborted.

4) It was about 10 minutes hung up with the .NET installer progress bar about 1/4 of the way done. I might have been impatient on that.

5) Even the installers for the previous versions I had (which used to work) don't work now.

**Comment by yser23482hat** [comment]
SOLVED

Decided to completely uninstall .NET 4.8 and reinstall it from the Microsoft installer. Now TL works. Maybe I had an incomplete or corrupt .NET installation somehow.

Thank you for your replies and help on this issue!

**Comment by dokoham** [comment]
Please help...it cannot install 4.8 framework 
every time unknow error

**Comment by dokoham** [comment]
[2200:26F0][2019-08-25T03:43:27]i001: Burn v3.11.1.2318, Windows v10.0 (Build 17134: Service Pack 0), path: C:\Users\ADMINI~1\AppData\Local\Temp\{B636F12E-6BC3-463A-BF88-21F4E37FD58F}\.cr\TwitchLeecher_1.7.exe
[2200:26F0][2019-08-25T03:43:27]i000: Initializing string variable 'BUNDLE_MANUFACTURER' to value 'Franiac'
[2200:26F0][2019-08-25T03:43:27]i000: Initializing string variable 'BUNDLE_PRODUCT_NAME' to value 'Twitch Leecher'
[2200:26F0][2019-08-25T03:43:27]i000: Initializing string variable 'BUNDLE_PRODUCT_VERSION_PADDED' to value '1.7.0.0'
[2200:26F0][2019-08-25T03:43:27]i000: Initializing string variable 'BUNDLE_PRODUCT_VERSION_TRIMMED' to value '1.7'
[2200:26F0][2019-08-25T03:43:27]i009: Command Line: '-burn.clean.room=C:\Users\Administrator\Desktop\TwitchLeecher_1.7.exe -burn.filehandle.attached=660 -burn.filehandle.self=668'
[2200:26F0][2019-08-25T03:43:27]i000: Setting string variable 'WixBundleOriginalSource' to value 'C:\Users\Administrator\Desktop\TwitchLeecher_1.7.exe'
[2200:26F0][2019-08-25T03:43:27]i000: Setting string variable 'WixBundleOriginalSourceFolder' to value 'C:\Users\Administrator\Desktop\'
[2200:26F0][2019-08-25T03:43:27]i000: Setting string variable 'WixBundleLog' to value 'C:\Users\ADMINI~1\AppData\Local\Temp\Twitch_Leecher_1.7_20190825034327.log'
[2200:26F0][2019-08-25T03:43:27]i000: Setting string variable 'WixBundleName' to value 'Twitch Leecher 1.7'
[2200:26F0][2019-08-25T03:43:27]i000: Setting string variable 'WixBundleManufacturer' to value 'Franiac'
[2200:26F0][2019-08-25T03:43:27]i000: Loading prerequisite bootstrapper application because managed host could not be loaded, error: 0x80131700.
[2200:2890][2019-08-25T03:43:27]i000: Setting numeric variable 'WixStdBALanguageId' to value 2052
[2200:2890][2019-08-25T03:43:27]i000: Setting version variable 'WixBundleFileVersion' to value '1.7.0.0'
[2200:26F0][2019-08-25T03:43:27]i100: Detect begin, 2 packages
[2200:26F0][2019-08-25T03:43:27]i000: Setting string variable 'NetFxCurrent' to value '461808'
[2200:26F0][2019-08-25T03:43:27]i102: Detected related bundle: {8db3ac0e-bbba-4b04-a5d5-d9b353bb6030}, type: Upgrade, scope: PerMachine, version: 1.5.6.0, operation: MajorUpgrade
[2200:26F0][2019-08-25T03:43:27]w120: Detected partially cached package: NetFxWeb, invalid payload: NetFxWeb, reason: 0x80070002
[2200:26F0][2019-08-25T03:43:27]i052: Condition 'NetFxCurrent >= 528040' evaluates to false.
[2200:26F0][2019-08-25T03:43:27]i103: Detected related package: {4E2BCEDA-504B-4D80-8B59-7B30D634C3B9}, scope: PerMachine, version: 1.5.6.0, language: 0 operation: MajorUpgrade
[2200:26F0][2019-08-25T03:43:27]i101: Detected package: NetFxWeb, state: Absent, cached: Partial
[2200:26F0][2019-08-25T03:43:27]i101: Detected package: TL, state: Absent, cached: None
[2200:26F0][2019-08-25T03:43:27]i199: Detect complete, result: 0x0
[2200:26F0][2019-08-25T03:43:31]i200: Plan begin, 2 packages, action: Install
[2200:26F0][2019-08-25T03:43:31]i202: Planned bundle: {8db3ac0e-bbba-4b04-a5d5-d9b353bb6030}, ba requested state: None over default: Absent
[2200:26F0][2019-08-25T03:43:31]w321: Skipping dependency registration on package with no dependency providers: NetFxWeb
[2200:26F0][2019-08-25T03:43:31]i000: Setting string variable 'NetFxFullLog' to value 'C:\Users\ADMINI~1\AppData\Local\Temp\Twitch_Leecher_1.7_20190825034327_000_NetFxWeb.log'
[2200:26F0][2019-08-25T03:43:31]i201: Planned package: NetFxWeb, state: Absent, default requested: Present, ba requested: Present, execute: Install, rollback: None, cache: Yes, uncache: No, dependency: None
[2200:26F0][2019-08-25T03:43:31]i201: Planned package: TL, state: Absent, default requested: Present, ba requested: None, execute: None, rollback: None, cache: No, uncache: No, dependency: None
[2200:26F0][2019-08-25T03:43:31]i207: Planned related bundle: {8db3ac0e-bbba-4b04-a5d5-d9b353bb6030}, type: Upgrade, default requested: Absent, ba requested: None, execute: None, rollback: None, dependency: None
[2200:26F0][2019-08-25T03:43:31]i299: Plan complete, result: 0x0
[2200:26F0][2019-08-25T03:43:31]i300: Apply begin
[2200:26F0][2019-08-25T03:43:31]i010: Launching elevated engine process.
[2200:26F0][2019-08-25T03:43:33]i011: Launched elevated engine process.
[2200:26F0][2019-08-25T03:43:33]i012: Connected to elevated engine.
[2E50:2EB0][2019-08-25T03:43:33]i358: Pausing automatic updates.
[2E50:2EB0][2019-08-25T03:43:33]i359: Paused automatic updates.
[2E50:2EB0][2019-08-25T03:43:33]i360: Creating a system restore point.
[2E50:2EB0][2019-08-25T03:43:33]w363: Could not create system restore point, error: 0x80070422. Continuing...
[2E50:2EB0][2019-08-25T03:43:33]i370: Session begin, registration key: SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{d1c2c779-f1ef-41e0-926a-443b289eaefd}, options: 0x7, disable resume: No
[2E50:2EB0][2019-08-25T03:43:33]i000: Caching bundle from: 'C:\Users\ADMINI~1\AppData\Local\Temp\{4C012F04-AA7F-48DC-91C0-6B51568E6A47}\.be\TwitchLeecher.exe' to: 'C:\ProgramData\Package Cache\{d1c2c779-f1ef-41e0-926a-443b289eaefd}\TwitchLeecher.exe'
[2E50:2EB0][2019-08-25T03:43:33]i320: Registering bundle dependency provider: {d1c2c779-f1ef-41e0-926a-443b289eaefd}, version: 1.7.0.0
[2E50:2EB0][2019-08-25T03:43:33]i371: Updating session, registration key: SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{d1c2c779-f1ef-41e0-926a-443b289eaefd}, resume: Active, restart initiated: No, disable resume: No
[2200:03E0][2019-08-25T03:43:33]w343: Prompt for source of package: NetFxWeb, payload: NetFxWeb, path: C:\Users\Administrator\Desktop\ndp48-web.exe
[2200:03E0][2019-08-25T03:43:33]i338: Acquiring package: NetFxWeb, payload: NetFxWeb, download from: http://go.microsoft.com/fwlink/?LinkId=2085155
[2E50:2738][2019-08-25T03:43:37]e000: Error 0x80091007: Hash mismatch for path: C:\ProgramData\Package Cache\.unverified\NetFxWeb, expected: 755349ECD6A478FE010E466B29911D2388F6CE94, actual: 5A84A8E612E270E27D0061D58DB6B470153BE1F9
[2E50:2738][2019-08-25T03:43:37]e000: Error 0x80091007: Failed to verify hash of payload: NetFxWeb
[2E50:2738][2019-08-25T03:43:37]e310: Failed to verify payload: NetFxWeb at path: C:\ProgramData\Package Cache\.unverified\NetFxWeb, error: 0x80091007. Deleting file.
[2E50:2738][2019-08-25T03:43:37]e000: Error 0x80091007: Failed to cache payload: NetFxWeb
[2200:03E0][2019-08-25T03:43:37]e314: Failed to cache payload: NetFxWeb from working path: C:\Users\ADMINI~1\AppData\Local\Temp\{4C012F04-AA7F-48DC-91C0-6B51568E6A47}\NetFxWeb, error: 0x80091007.
[2200:03E0][2019-08-25T03:43:37]e349: Application requested retry of payload: NetFxWeb, encountered error: 0x80091007. Retrying...
[2200:03E0][2019-08-25T03:43:37]w343: Prompt for source of package: NetFxWeb, payload: NetFxWeb, path: C:\Users\Administrator\Desktop\ndp48-web.exe
[2200:03E0][2019-08-25T03:43:40]i338: Acquiring package: NetFxWeb, payload: NetFxWeb, download from: http://go.microsoft.com/fwlink/?LinkId=2085155
[2E50:2738][2019-08-25T03:43:43]e000: Error 0x80091007: Hash mismatch for path: C:\ProgramData\Package Cache\.unverified\NetFxWeb, expected: 755349ECD6A478FE010E466B29911D2388F6CE94, actual: 5A84A8E612E270E27D0061D58DB6B470153BE1F9
[2E50:2738][2019-08-25T03:43:43]e000: Error 0x80091007: Failed to verify hash of payload: NetFxWeb
[2E50:2738][2019-08-25T03:43:43]e310: Failed to verify payload: NetFxWeb at path: C:\ProgramData\Package Cache\.unverified\NetFxWeb, error: 0x80091007. Deleting file.
[2E50:2738][2019-08-25T03:43:43]e000: Error 0x80091007: Failed to cache payload: NetFxWeb
[2200:03E0][2019-08-25T03:43:43]e314: Failed to cache payload: NetFxWeb from working path: C:\Users\ADMINI~1\AppData\Local\Temp\{4C012F04-AA7F-48DC-91C0-6B51568E6A47}\NetFxWeb, error: 0x80091007.
[2200:03E0][2019-08-25T03:43:43]e349: Application requested retry of payload: NetFxWeb, encountered error: 0x80091007. Retrying...
[2200:03E0][2019-08-25T03:43:43]w343: Prompt for source of package: NetFxWeb, payload: NetFxWeb, path: C:\Users\Administrator\Desktop\ndp48-web.exe
[2200:03E0][2019-08-25T03:43:46]i338: Acquiring package: NetFxWeb, payload: NetFxWeb, download from: http://go.microsoft.com/fwlink/?LinkId=2085155
[2E50:2738][2019-08-25T03:43:48]e000: Error 0x80091007: Hash mismatch for path: C:\ProgramData\Package Cache\.unverified\NetFxWeb, expected: 755349ECD6A478FE010E466B29911D2388F6CE94, actual: 5A84A8E612E270E27D0061D58DB6B470153BE1F9
[2E50:2738][2019-08-25T03:43:48]e000: Error 0x80091007: Failed to verify hash of payload: NetFxWeb
[2E50:2738][2019-08-25T03:43:48]e310: Failed to verify payload: NetFxWeb at path: C:\ProgramData\Package Cache\.unverified\NetFxWeb, error: 0x80091007. Deleting file.
[2E50:2738][2019-08-25T03:43:48]e000: Error 0x80091007: Failed to cache payload: NetFxWeb
[2200:03E0][2019-08-25T03:43:48]e314: Failed to cache payload: NetFxWeb from working path: C:\Users\ADMINI~1\AppData\Local\Temp\{4C012F04-AA7F-48DC-91C0-6B51568E6A47}\NetFxWeb, error: 0x80091007.
[2200:26F0][2019-08-25T03:43:48]e000: Error 0x80091007: Failed while caching, aborting execution.
[2E50:2EB0][2019-08-25T03:43:48]i372: Session end, registration key: SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{d1c2c779-f1ef-41e0-926a-443b289eaefd}, resume: None, restart: None, disable resume: No
[2E50:2EB0][2019-08-25T03:43:48]i330: Removed bundle dependency provider: {d1c2c779-f1ef-41e0-926a-443b289eaefd}
[2E50:2EB0][2019-08-25T03:43:48]i352: Removing cached bundle: {d1c2c779-f1ef-41e0-926a-443b289eaefd}, from path: C:\ProgramData\Package Cache\{d1c2c779-f1ef-41e0-926a-443b289eaefd}\
[2E50:2EB0][2019-08-25T03:43:48]i371: Updating session, registration key: SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{d1c2c779-f1ef-41e0-926a-443b289eaefd}, resume: None, restart initiated: No, disable resume: No
[2200:26F0][2019-08-25T03:43:48]i399: Apply complete, result: 0x80091007, restart: None, ba requested restart:  No

**Comment by LeMageFro** [comment]
> Please help...it cannot install 4.8 framework
> every time unknow error

I had a similar problem, try installing .NET 4.8 manually from this link : https://dotnet.microsoft.com/download/dotnet-framework/net48
After that TL 1.7 installer should work.
